libs/SalesforceSDKCore/SalesforceSDKCoreTests/SFUserAccountManagerTests.m

```diff
@@ -186,22 +186,15 @@ - (void)testSingleAccount {
     NSArray *accounts = [self createAndVerifyUserAccounts:1];
     SFUserAccount *user = accounts[0];
     // Check if the UserAccount.plist is stored at the right location
-    NSError *error = nil;
-    XCTAssertTrue([self.uam saveAccounts:&error], @"Unable to save user accounts: %@", error);
-    
     NSString *expectedLocation = [[SFDirectoryManager sharedManager] directoryForOrg:user.credentials.organizationId user:user.credentials.userId community:nil type:NSLibraryDirectory components:nil];
     expectedLocation = [expectedLocation stringByAppendingPathComponent:@"UserAccount.plist"];
-    XCTAssertEqualObjects(expectedLocation, [SFUserAccountManager userAccountPlistFileForUser:user], @"Mismatching user account paths");
+    XCTAssertEqualObjects(expectedLocation, [SFDefaultUserAccountPersister userAccountPlistFileForUser:user], @"Mismatching user account paths");
     NSFileManager *fm = [[NSFileManager alloc] init];
     XCTAssertTrue([fm fileExistsAtPath:expectedLocation], @"Unable to find new UserAccount.plist");
-    
-    // Now remove all the users and re-load
-    [self.uam clearAllAccountState];
-    XCTAssertEqual([self.uam.allUserIdentities count], (NSUInteger)0, @"There should be no accounts");
 
-    XCTAssertTrue([self.uam loadAccounts:&error], @"Unable to load user accounts: %@", error);
     NSString *userId = [NSString stringWithFormat:kUserIdFormatString, (unsigned long)0];
     XCTAssertEqualObjects(((SFUserAccountIdentity *)self.uam.allUserIdentities[0]).userId, userId, @"User ID doesn't match after reload");
+     [self deleteUserAndVerify:user userDir:expectedLocation];
 }
 
 - (void)testMultipleAccounts {
```

Comment:
Okay, here's what I think about this test, now that I've traipsed around in it a bit.  This test's most useful purpose is actually as a test of `SFUserAccountPersister` functionality—I guess specifically the default persister, where accounts are stored and retrieved from disk.

I think this test should be reworked in that specific isolated context (i.e. as a `SFDefaultUserAccountPersister`-specific test), and otherwise removed from here.  There's a lot of baggage around `SFUserAccountManager` here that's not actually accomplishing anything meaningful.